packages/dropdowns/src/Dropdown/Dropdown.tsx

```diff
@@ -7,18 +7,42 @@
 
 import React, { useRef } from 'react';
 import PropTypes from 'prop-types';
-import Downshift from 'downshift';
+import Downshift, { ControllerStateAndHelpers, StateChangeOptions } from 'downshift';
 import { Manager } from 'react-popper';
 import { withTheme, isRtl } from '@zendeskgarden/react-theming';
 import { KEY_CODES, composeEventHandlers } from '@zendeskgarden/container-selection';
 
-export const DropdownContext = React.createContext();
+export interface IDropdownContext {
+  itemIndexRef: React.MutableRefObject<number>;
+  previousItemRef: React.MutableRefObject<any>;
+  previousIndexRef: React.MutableRefObject<any>;
+  nextItemsHashRef: React.MutableRefObject<object>;
+  popperReferenceElementRef: React.MutableRefObject<any>;
+  selectedItems?: any[];
+  downshift: ControllerStateAndHelpers<any>;
+}
+
+export const DropdownContext = React.createContext<IDropdownContext | undefined>(undefined);
+
+export interface IDropdownProps {
+  isOpen?: boolean;
+  selectedItem?: any;
+  selectedItems?: any[];
+  highlightedIndex?: number;
+  inputValue?: string;
+  onSelect?: (selectedItem: any | null, stateAndHelpers: ControllerStateAndHelpers<any>) => void;
+  onStateChange?: (
+    options: StateChangeOptions<any>,
+    stateAndHelpers: ControllerStateAndHelpers<any>
+  ) => void;
+  downshiftProps?: object;
+}
```

Comment:
Should we have a bit more organisation around where interfaces should live? Should it be in the file itself or live along side it in another file?